cli/drivers/WordpressComposerInstallValetDriver.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+class WordpressComposerInstallValetDriver extends ValetDriver
+{
+    /**
+     * Determine if the driver serves the request.
+     *
+     * @param  string  $sitePath
+     * @param  string  $siteName
+     * @param  string  $uri
+     * @return void
+     */
+    public function serves($sitePath, $siteName, $uri)
+    {
+        if (file_exists($sitePath . '/composer.json')
+            && (file_exists($sitePath . '/wp-config.php')
+                || $this->isJohnpblochInstall($sitePath)
+                || $this->isWebrootInstall($sitePath))
+        ) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Determine if the incoming request is for a static file.
+     *
+     * @param  string  $sitePath
+     * @param  string  $siteName
+     * @param  string  $uri
+     * @return string|false
+     */
+    public function isStaticFile($sitePath, $siteName, $uri)
+    {
+        // Look in public web directory
+        if ($this->isBedrockInstall($sitePath)) {
+            $sitePath .= '/web';
+        }
+
+        if ($this->getStaticFilePath($sitePath, $uri)) {
+            return $this->getStaticFilePath($sitePath, $uri);
+        }
+
+        if ($this->isWebrootInstall($sitePath)) {
+            $sitePath = $this->getWebrootInstallPath($sitePath);
+
+            if ($this->getStaticFilePath($sitePath, $uri)) {
+                return $this->getStaticFilePath($sitePath, $uri);
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Get the fully resolved path to the application's front controller.
+     *
+     * @param  string  $sitePath
+     * @param  string  $siteName
+     * @param  string  $uri
+     * @return string
+     */
+    public function frontControllerPath($sitePath, $siteName, $uri)
+    {
+        // Look in public web directory
+        if ($this->isBedrockInstall($sitePath)) {
```

Comment:
As a generic driver, this should probably not handle specific cases like Bedrock, IMHO.
Specific implementations are better off with their own driver.